enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,974 @@
+     [...]
+   ```
+
+#### Configuring Auto Subnet Selection for IngressControllers with BYO VPC during Installation Workflow
+
+A cluster admin wants to provide cluster-role subnets to use an existing VPC, but still have the IngressController
+automatically discover the subnets for its load balancer: 
+
+1. Cluster admin creates an install-config with `platform.aws.subnetsConfig` specified
+   with the subnet(s) while ensuring `platform.aws.subnetsConfig[].roles` is configured
+   with only `Cluster`:
+    ```yaml
+    apiVersion: v1
+    baseDomain: devcluster.openshift.com
+    metadata:
+      name: my-cluster
+    [...]
+    platform:
+      aws:
+        region: us-east-2
+        subnetsConfig:
+        - id: subnet-0fcf8e0392f0910d2
+          roles:
+          - Cluster
+        - id: subnet-0fcf8e0392f0910d3
+          roles:
+          - Cluster
+    [...]
+    ```
+2. The OpenShift Installer installs the cluster into the VPC where the cluster-role subnets exist, but does not
+   configure any specific subnets for the `default` IngressController or set defaults for new IngressControllers.
+3. When the `default` IngressController is created, the load balancer will automatically map to the appropriately
+   tagged subnets in the VPC.
+
+#### Specifying Dedicated IngressController-role Subnets During Installation with a BYO VPC Workflow
+
+Specifying dedicated IngressController subnets enables cluster admins to separate
+IngressController subnets from the node subnets (i.e. cluster-role subnets):
+
+1. Cluster admin creates an install-config with `platform.aws.subnetsConfig`, specifying
+   each subnet to have either the `IngressController` or `Cluster` role in `platform.aws.subnetsConfig.roles`, but
+   not both. Note that each `IngressController` role subnet MUST have a corresponding `Cluster` role subnet in the
+   same AZ; otherwise, the load balancer will be unable to route traffic into the cluster: 
+    ```yaml
+    apiVersion: v1
+    baseDomain: devcluster.openshift.com
+    metadata:
+      name: my-cluster
+    [...]
+    platform:
+      aws:
+        region: us-east-2
+        subnetsConfig:
+        - id: subnet-0fcf8e0392f0910d4 # AZ us-east-1
+          roles:
+          - Cluster
+        - id: subnet-0fcf8e0392f0910d5 # AZ us-east-1
+          roles:
+          - Ingress
+        lbType: NLB
+    [...]
+    ```
+2. The OpenShift Installer installs the cluster into the VPC where the subnets exist, and populates the
+   IngressConfig's `spec.loadBalancer.platform.aws.networkLoadBalancer.subnets` and
+   `spec.loadBalancer.platform.aws.networkLoadBalancer.subnets` field.
+3. When the Ingress Operator starts, it uses the IngressConfig's `Subnets` value for the `default` IngressController
+   as well as the default for all new IngressControllers.
+4. Since the IngressController subnet(s) are in the same AZ as the cluster-role subnets(s), ingress traffic is successfully
+   routed from the load balancer to the nodes.
```

Comment:
What will be the [health check state][state] of the targets attached to the Load Balancers in this use case where the subnets of nodes are different than the subnets of LB in the same AZ? Will CLB and NLB have the same routing behavior?

[state]: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-health-checks.html#target-health-states